src/fclone.js

```diff
@@ -20,7 +20,7 @@ function fclone(obj, refs) {
   }
 
   // typed array Int32Array etc.
-  if (/\d[a-z]*Array]$/i.test(Object.prototype.toString.call(obj))) {
+  if (obj.subarray  && typeof obj.subarray === 'function' && /^[A-Z][A-Za-z\d]+Array$/.test(Object.prototype.toString.call(obj))) {
```

Comment:
What was wrong with the old regex? 

Btw `typeof obj.subarray === 'function'` is sufficient, no need for `obj.subarray`